Extensions/Effects/JsExtension.js

```diff
@@ -27,127 +27,289 @@ module.exports = {
+        .setType("resource")
+        .addExtraInfo("image")
+        .setLabel(_("Displacement map texture for the effect"))
+    );
+    displacementProperties.set(
+      "scaleX",
+      new gd.PropertyDescriptor(/* defaultValue= */ "20")
+        .setLabel(_("Scale X"))
+        .setType("number")
+    );
+    displacementProperties.set(
+      "scaleY",
+      new gd.PropertyDescriptor(/* defaultValue= */ "20")
+        .setLabel(_("Scale Y"))
+        .setType("number")
+    );
+
+    const colorMapEffect = extension
+      .addEffect("ColorMap")
+      .setFullName(_("Color Map"))
+      .setDescription(_("Add Color Map effect"))
+      .addIncludeFile("Extensions/Effects/color-map-pixi-filter.js")
+      .addIncludeFile("Extensions/Effects/pixi-filters/filter-color-map.js");
+    const colorMapProperties = colorMapEffect.getProperties();
+    colorMapProperties.set(
+      "colorMapTexture",
+      new gd.PropertyDescriptor("")
+        .setType("resource")
+        .addExtraInfo("image")
+        .setLabel(_("Color map texture for the effect"))
+        .setDescription(
+          _(
+            "Color map are like LUT, you can change colors of pixels by modifing a reference color image containing each colours. Orignal map file can be found at [TODO INSERT PATH]"
+          )
+        )
+    );
+    colorMapProperties.set(
+      "nearest",
+      new gd.PropertyDescriptor(/* defaultValue= */ "false")
+        .setLabel(_("Nearest"))
+        .setType("boolean")
+        .setDescription(_("Enable antialiasing on gradients."))
+    );
+    colorMapProperties.set(
+      "mix",
+      new gd.PropertyDescriptor(/* defaultValue= */ "100")
+        .setLabel(_("Mix"))
+        .setType("number")
+        .setDescription(
+          _("Mix value of the effect on the layer. (in percent)")
+        )
+    );
+
+    const pixelateEffect = extension
+      .addEffect("Pixelate")
+      .setFullName(_("Pixelate"))
+      .setDescription(_("Pixelize the layer"))
+      .addIncludeFile("Extensions/Effects/pixelate-pixi-filter.js")
+      .addIncludeFile("Extensions/Effects/pixi-filters/filter-pixelate.js");
+    const pixelateProperties = pixelateEffect.getProperties();
+
+    pixelateProperties.set(
+      "size",
+      new gd.PropertyDescriptor(/* defaultValue= */ "10")
+        .setLabel(_("Size"))
+        .setType("number")
+        .setDescription(_("Size of pixel effect (10 pixels by default)"))
+    );
+
+    const reflectionEffect = extension
+      .addEffect("Reflection")
+      .setFullName(_("Reflection"))
+      .setDescription(_("Reflect the layer"))
+      .addIncludeFile("Extensions/Effects/reflection-pixi-filter.js")
+      .addIncludeFile("Extensions/Effects/pixi-filters/filter-reflection.js");
+    const reflectionProperties = reflectionEffect.getProperties();
+
+    reflectionProperties.set(
+      "mirror",
+      new gd.PropertyDescriptor(/* defaultValue= */ "true")
+        .setLabel(_("mirror"))
```

Comment:
The label is shown in the UI to the user. It should be what is displayed next to the checkbox, so more something that looks like the description: "Reflect the image on the waves"